pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalanceIntegrationTest.java

```diff
@@ -1379,6 +1388,284 @@ public void testRebalanceJobZkMetadataCleanup()
+          .keySet();
+      assertFalse(originalConsumingSegmentsToMove.isEmpty());
+      assertEquals(
+          rebalanceResult.getRebalanceSummaryResult().getServerInfo().getNumServers().getExpectedValueAfterRebalance(),
+          2);
+
+      waitForRebalanceToComplete(rebalanceResult.getJobId(), 30000);
+
+      response = sendGetRequest(getControllerRequestURLBuilder().forTableConsumingSegmentsInfo(getTableName()));
+      consumingSegmentInfoResponse =
+          JsonUtils.stringToObject(response, ConsumingSegmentInfoReader.ConsumingSegmentsInfoMap.class);
+      consumingSegmentNow = new LLCSegmentName(
+          consumingSegmentInfoResponse._segmentToConsumingInfoMap.keySet().stream().sorted().iterator().next());
+      consumingSegmentOriginal =
+          new LLCSegmentName(originalConsumingSegmentsToMove.stream().sorted().iterator().next());
+      assertEquals(consumingSegmentNow.getSequenceNumber(), consumingSegmentOriginal.getSequenceNumber() + 1);
+
+      // test: move segment from tenantA to tenantB with includeConsuming = false, consuming segment should not be
+      // committed
+
+      tableConfig.setTenantConfig(new TenantConfig(getBrokerTenant(), tenantB, null));
+      updateTableConfig(tableConfig);
+
+      rebalanceConfig.setForceCommit(true);
+      rebalanceConfig.setDowntime(false);
+      rebalanceConfig.setIncludeConsuming(false);
+
+      response = sendPostRequest(getRebalanceUrl(rebalanceConfig, TableType.REALTIME));
+      rebalanceResult = JsonUtils.stringToObject(response, RebalanceResult.class);
+
+      waitForRebalanceToComplete(rebalanceResult.getJobId(), 30000);
+
+      response = sendGetRequest(getControllerRequestURLBuilder().forTableConsumingSegmentsInfo(getTableName()));
+      consumingSegmentInfoResponse =
+          JsonUtils.stringToObject(response, ConsumingSegmentInfoReader.ConsumingSegmentsInfoMap.class);
+      consumingSegmentOriginal = consumingSegmentNow;
+      consumingSegmentNow = new LLCSegmentName(
+          consumingSegmentInfoResponse._segmentToConsumingInfoMap.keySet().stream().sorted().iterator().next());
+      // the sequence number should not increase since the consuming segment is not committed
+      assertEquals(consumingSegmentNow.getSequenceNumber(), consumingSegmentOriginal.getSequenceNumber());
+
+      // Resume the table
+      tableConfig.setTenantConfig(new TenantConfig(getBrokerTenant(), getServerTenant(), null));
+      tableConfig.getValidationConfig().setReplication("1");
+      updateTableConfig(tableConfig);
+      rebalanceConfig.setForceCommit(false);
+      rebalanceConfig.setMinAvailableReplicas(0);
+      rebalanceConfig.setDowntime(false);
+      rebalanceConfig.setIncludeConsuming(true);
+      response = sendPostRequest(getRebalanceUrl(rebalanceConfig, TableType.REALTIME));
+      rebalanceResult = JsonUtils.stringToObject(response, RebalanceResult.class);
+      waitForRebalanceToComplete(rebalanceResult.getJobId(), 30000);
```

Comment:
IMO this should be in the finally block as well, since the other tests probably won't pass with the table in an unexpected tenant.